hledger-lib/Hledger/Read/Common.hs

```diff
@@ -870,21 +871,25 @@ fromRawNumber raw mExp = case raw of
 
   where
     -- Outputs digit group sizes from least significant to most significant
-    groupSizes :: [DigitGrp] -> [Int]
+    groupSizes :: [DigitGrp] -> [Word8]
     groupSizes digitGrps = reverse $ case map digitGroupLength digitGrps of
       (a:b:cs) | a < b -> b:cs
       gs               -> gs
 
-    toQuantity :: DigitGrp -> DigitGrp -> (Quantity, Int)
+    toQuantity :: DigitGrp -> DigitGrp -> (Quantity, Word8)
     toQuantity preDecimalGrp postDecimalGrp = (quantity, precision)
       where
-        quantity = Decimal (fromIntegral precision)
+        quantity = Decimal precision
                            (digitGroupNumber $ preDecimalGrp <> postDecimalGrp)
         precision = digitGroupLength postDecimalGrp
 
-    applyExp :: Int -> (Decimal, Int) -> (Decimal, Int)
-    applyExp exponent (quantity, precision) =
-      (quantity * 10^^exponent, max 0 (precision - exponent))
+    applyExp :: Integer -> (Decimal, Word8) -> (Decimal, Word8)
+    applyExp exponent (quantity, precision) = (quantity * 10^^exponent, newPrecision)
+      where
+        newPrecision | precisionDiff >= 255 = maxBound
+                     | precisionDiff <= 0   = 0
+                     | otherwise            = fromInteger precisionDiff
```

Comment:
and if precisionDiff is 254, that's maxprecision ? shouldn't we avoid assigning those special values as a result of parsing numbers ? So the max number of decimal places we should allow
 is 253 ?

Reply:
Again, superceded with later code.